imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,197 @@
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    with open(packet_path) as text_file, open(binary_file_path, "wb") as binary_file:
+        for line in text_file:
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1:]
+                # Convert hex to binary
+                binary_data = bytearray.fromhex("".join(hex_values))
+                binary_file.write(binary_data)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
+
+
+@pytest.fixture(scope="session")
+def empty_binary_packet_path():
+    """
+    Creates an empty binary file, representative of a scenario
+    in which no packet data is available.
+    """
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
+
+
+@pytest.fixture(scope="session")
+def decom_packets_data(binary_packet_path, xtce_ialirt_path):
+    """Read packet data from file using decom_packets"""
+    data_packet_list = decom_packets(binary_packet_path, xtce_ialirt_path)
+    return data_packet_list
+
+
+def test_length(decom_packets_data):
+    """Test if total packets in data file is correct"""
+    total_packets = 32
+    assert len(decom_packets_data) == total_packets
+
+
+def test_binary_value_length():
+    """
+    Validates the length of binary data converted
+    from a text file containing hexadecimal packet data.
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    with open(packet_path) as text_file, open(binary_file_path, "wb"):
+        for line in text_file:
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1:]
+                # Convert hex to binary
+                binary_data = bytearray.fromhex("".join(hex_values))
+                assert len(binary_data) * 8 == IALIRT_PACKET_LENGTH
+
+
+def test_generate_xarray_empty_file(empty_binary_packet_path, xtce_ialirt_path, caplog):
+    """Test that an error is logged if an empty file is passed to generate_xarray."""
+
+    result = generate_xarray(empty_binary_packet_path, xtce_ialirt_path)
+
+    assert "Error during packet decomposition" in caplog.text
+    assert result is None
```

Comment:
```suggestion
def test_generate_xarray_empty_file(empty_binary_packet_path, xtce_ialirt_path, caplog):
    """Test that an error is logged if an empty file is passed to generate_xarray."""
    with pytest.raises(ValueError):
        generate_xarray(empty_binary_packet_path, xtce_ialirt_path)
```
If you raise instead of log, we can assert that this does raise as expected here like this.